libbeat/idxmgmt/idxmgmt.go

```diff
@@ -113,54 +112,34 @@ func MakeDefaultSupport(ilmSupport lifecycle.SupportFactory) SupportFactory {
 			log = log.Named(logName)
 		}
 
-		// first fetch the ES output, check if we're running against serverless, use that to set a default config.
-		// the Supporter only cares about lifecycle config for checking if ILM/DSL is enabled or disabled.
-		outCfg := struct {
-			Output config.Namespace `config:"output"`
+		// now that we have the "correct" default, unpack the rest of the config
+		cfg := struct {
+			Lifecycle lifecycle.RawConfig `config:",inline"`
+			Template  *config.C           `config:"setup.template"`
+			Output    config.Namespace    `config:"output"`
+			Migration *config.C           `config:"migration.6_to_7"`
 		}{}
 		if configRoot != nil {
-			err := configRoot.Unpack(&outCfg)
-			if err != nil {
-				return nil, fmt.Errorf("error unpacking output config while making index support: %w", err)
-			}
-		}
-
-		defaultLifecycle := lifecycle.DefaultILMConfig(info)
-		if outCfg.Output.IsSet() && outCfg.Output.Name() == "elasticsearch" {
-			esClient, err := eslegclient.NewConnectedClient(outCfg.Output.Config(), info.Beat)
-			if err != nil {
-				// this is for the benefit of tests as well as various retry mechanisms;
-				// if we can't connect to ES, print a warning and fallback to ILM, tests won't care and non-test environments can
-				// retry the connection
-				logp.L().Warnf("could not connect to ES to determine ES type during index setup. Will default to ILM.")
-			} else {
-				if esClient.IsServerless() {
-					defaultLifecycle = lifecycle.DefaultDSLConfig(info)
-				}
+			if err := configRoot.Unpack(&cfg); err != nil {
+				return nil, fmt.Errorf("error unpacking cfg settings while setting up index support: %w", err)
 			}
-
 		}
 
-		// now that we have the "correct" default, unpack the rest of the config
-		cfg := struct {
-			ILM       lifecycle.LifecycleConfig `config:",inline"`
-			Template  *config.C                 `config:"setup.template"`
-			Output    config.Namespace          `config:"output"`
-			Migration *config.C                 `config:"migration.6_to_7"`
-		}{
-			ILM: defaultLifecycle,
+		// consider lifecycles enabled if the user has explicitly enabled them,
+		// or if no `enabled` setting has been set by the user, thus reverting to a default of enabled.
+		enabled := false
+		if cfg.Lifecycle.DSL.Enabled() || cfg.Lifecycle.ILM.Enabled() {
+			enabled = true
 		}
-		if configRoot != nil {
-			if err := configRoot.Unpack(&cfg); err != nil {
-				return nil, err
-			}
+		if (cfg.Lifecycle.DSL == nil || !cfg.Lifecycle.DSL.HasField("enabled")) && (cfg.Lifecycle.ILM == nil || !cfg.Lifecycle.ILM.HasField("enabled")) {
+			enabled = true
```

Comment:
"(if DSL is nil OR not enabled) AND (ILM is nil OR not enabled) then lifecycle is enabled" reads like the condition is backwards. If there are not allocated objects and nothing is enabled, why would we set enabled = true?

Reply:
So, my assumption, based on [older versions](https://github.com/elastic/beats/blob/231e93a6d20e35a8fc6f34dc767110b8a27ec645/libbeat/idxmgmt/ilm/config.go#L69) is that the lifecycle management should default to enabled if the user hasn't explicitly configured it.

Reply:
Ah this needs a comment then explaining what is happening and why, in isolation it looks wrong.